src/puppetlabs/puppetdb/queue.clj

```diff
@@ -179,7 +184,10 @@
                                       tcoerce/from-long)
                  :version (Long/parseLong version)
                  :command (get md-cmd->pdb-cmd md-command "unknown")
-                 :certname certname})))))))
+                 :certname certname
+                 :compression (if (nil? compression)
+                                ""
+                                (subs compression 1))})))))))
```

Comment:
This code just parses back the file extension as stockpile would store it to disk.  Since it's stored as `.json.gz`, this strips it to `gz` in the conversion back to a command.  The PR currently only converts from a `content-encoding` (which would be "gzip") to a file extension (which would be "gz") when the command is first dumped into the queue from the HTTP route handling, in `enqueue-command-handler`.  I could store the payload to disk as `.json.gzip` to avoid the conversion, but then we'd be using even more characters in memory.  This might also look weird from the perspective of someone inspecting files on the file system, too, since `...gz` seems more standard.